torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -951,6 +833,154 @@ def _should_free_in_backward(
     )
 
 
```

Comment:
Are you perhaps willing to add small top level comments for each of the new functions? :)

Reply:
I added the comments in the subsequent PR since the async HSDP refactors these a bit more now.